tools/zip.js

```diff
@@ -38,15 +39,25 @@ function createZip() {
     });
 
     archive.pipe(file);
-    archive.glob('**', {ignore: exclude});
+    if (isFirefox) {
+      const name = 'manifest.json';
+      const keyOpt = 'optional_permissions';
+      ignore.unshift(name);
+      manifest[keyOpt] = manifest[keyOpt].filter(p => p !== 'declarativeContent');
+      if (!manifest[keyOpt].length) {
+        delete manifest[keyOpt];
+      }
+      archive.append(JSON.stringify(manifest, null, '  '), {name, stats: fs.lstatSync(name)});
+    }
+    archive.glob('**', {ignore});
     // Don't use modified codemirror.js (see "update-libraries.js")
     archive.directory('node_modules/codemirror/lib', 'vendor/codemirror/lib');
     archive.finalize();
   });
 }
 
-createZip()
-  .then(() => console.log('\x1b[32m%s\x1b[0m', 'Stylus zip complete'))
-  .catch(err => {
-    throw err;
-  });
+(async () => {
+  await createZip();
+  await createZip({isFirefox: true});
+  console.log('\x1b[32m%s\x1b[0m', 'Stylus zip complete');
+})();
```

Comment:
We still need a catch at the end or we will get an `UnhandledRejection` error when the async function throws:
```js
(...)().catch(err => {
  console.error(err);
  process.exit(1);
})
```

Reply:
What is difference in behavior? Previously `catch` simply forwarded the exception via `throw` so why should we manually print it and set the exit code?

Reply:
Aha, doing it explicitly ourselves is the only way to avoid the deprecation warnings spit by nodejs.